.github/workflows/ci-mlir.yml

```diff
@@ -18,98 +18,105 @@ jobs:
-        cd xdsl
-        # Add the MLIR Python bindings to the PYTHONPATH
-        export PYTHONPATH=$PYTHONPATH:${GITHUB_WORKSPACE}/llvm-project/build/tools/mlir/python_packages/mlir_core
-        pytest --cov --cov-config=.coveragerc .
-
-    - name: Execute lit tests
-      run: |
-        cd xdsl
-        export PYTHONPATH=$(pwd)
-        # Add mlir-opt to the path
-        export PATH=$PATH:${GITHUB_WORKSPACE}/llvm-project/build/bin/
-        # Add the MLIR Python bindings to the PYTHONPATH
-        export PYTHONPATH=$PYTHONPATH:${GITHUB_WORKSPACE}/llvm-project/build/tools/mlir/python_packages/mlir_core
-        lit -v tests/filecheck/ -DCOVERAGE -DEXEC_DIR=$(pwd)
-
-    - name: Combine coverage data
-      run: |
-        cd xdsl
-        coverage combine --append
-        coverage report
-        coverage xml
-
-    - name: Upload coverage to Codecov
-      if: matrix.python-version == '3.10'
-      uses: codecov/codecov-action@v3
-      with:
-        fail_ci_if_error: true
-        verbose: true
-        directory: ${GITHUB_WORKSPACE}/../
-        files: coverage.xml
-        token: ${{ secrets.CODECOV_TOKEN }}
+      - uses: actions/checkout@v3
+        with:
+          path: xdsl
+
+      - name: Python Setup
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Upgrade pip
+        run: |
+          pip install --upgrade pip
+
+      - name: Install the package locally
+        run: |
+          # Change directory so that xdsl-opt can be found during installation.
+          cd xdsl
+          pip install -e .[extras]
+
+      - name: Install CI dependencies
+        run: |
+          pip install codecov
+
+      - name: Cache binaries
+        id: cache-binary
+        uses: actions/cache@v3
+        with:
+          path: llvm-project/build
+          key: binaries-${{ runner.os }}-${{ env.MLIR-Version }}
+          restore-keys: binaries-${{ runner.os }}-${{ env.MLIR-Version }}
+          append-timestamp: false
+
+
+      - name: Checkout MLIR
+        if: steps.cache-binary.outputs.cache-hit != 'true'
+        uses: actions/checkout@v3
+        with:
+          repository: llvm/llvm-project.git
+          path: llvm-project
+          ref: ${{ env.MLIR-Version }}
+
+      - name: Clang Setup
+        if: steps.cache-binary.outputs.cache-hit != 'true'
+        uses: egor-tensin/setup-clang@v1
+
+      - name: Ninja Setup
+        if: steps.cache-binary.outputs.cache-hit != 'true'
+        uses: lukka/get-cmake@9e431acfe656e5db66cd4930386328fce59cfaba
+
+      - name: MLIR Build Setup
+        if: steps.cache-binary.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p llvm-project/build
+          cd llvm-project/build
+          cmake -G Ninja ../llvm -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_TARGETS_TO_BUILD="X86" -DLLVM_ENABLE_LLD=ON -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=ON
```

Comment:
should we split each argument to a separate line?

Reply:
Not sure whether I did this correctly. Should we just see after the merge or should I retrigger the CI? What do you prefer?

Reply:
I retriggered and it seems to work.